### PR TITLE
Support zero-downtime migration between ES clusters

### PR DESCRIPTION
**Pull request by akshaymankar**
https://wearezeta.atlassian.net/wiki/spaces/PS/pages/548798616/ElasticSearch+Migration+Proposal#3.-Zero-Downtime-Migration%3F

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

File: docs/reference/elastic-search.md [hunk 1]
```diff
@@ -117,7 +117,57 @@ deleted from the index. Attempts at reproducing this issue in a simpler
 environment have failed. As a workaround, there is a tool in
 [tools/db/find-undead](../../tools/db/find-undead) which can be used to find the
 undead users right after the migration. If they exist, please run refill the ES
-documents from cassandra as described [above](#refill-es-documents-from-cassandra)
+documents from cassandra as described [above](#refill-es-documents-from-cassandra**
```
Comment (jschaul) [inline docs/reference/elastic-search.md:120]:
```suggestion
documents from cassandra as described [above](#refill-es-documents-from-cassandra)
```

File: docs/reference/elastic-search.md [hunk 2]
```diff
@@ -117,7 +117,57 @@ deleted from the index. Attempts at reproducing this issue in a simpler
...
+   docker run "quay.io/wire/brig-index:$WIRE_VERSION" create \
+       --elasticsearch-server "http://$ES_NEW_HOST:$ES_NEW_PORT" \
+       --elasticsearch-index "$ES_NEW_INDEX" \
+       --elasticsearch-shards "$SHARDS" \
+       --elasticsearch-replicas "$REPLICAS" \
+       --elasticsearch-refresh-interval "$REFRESH_INTERVAL"
+   ```
+1. Redeploy brig with `elasticsearch.additionalWriteIndexUrl` set to the URL of
+   the new cluster and `elasticsearch.additionalWriteIndex` set to
+   `$ES_NEW_INDEX`. Make sure no old instances of brig are running.
```
Comment (jschaul) [inline docs/reference/elastic-search.md:157]: Should the "make sure no old instances of brig are running be a separate step, perhaps?"

File: docs/reference/elastic-search.md [hunk 3]
```diff
@@ -117,7 +117,57 @@ deleted from the index. Attempts at reproducing this issue in a simpler
...
+ES_OLD_HOST=<YOUR_HOST>
+ES_OLD_PORT=<YOUR_PORT> # usually 9200
+ES_OLD_INDEX=<INDEX_NAME_ALREADY_IN_USE>
+ES_NEW_HOST=<YOUR_HOST>
+ES_NEW_PORT=<YOUR_PORT> # usually 9200
+ES_NEW_INDEX=<NEW_INDEX_NAME>
+WIRE_VERSION=<VERSION_YOU_ARE_DEPLOYING>
+SHARDS=<NUMBER_OF_SHARDS_FOR_THE_NEW_INDEX>
+REPLICAS=<NUMBER_OF_REPLICAS_FOR_THE_INDEX>
+REFRESH_INTERVAL=<REFRESH_INTERVAL_IN_SECONDS>
```
Comment (jschaul) [inline docs/reference/elastic-search.md:140]: Can you document a command here which allows to see the current shards, replicas and refresh interval of the existing index (as I suppose one usually wants to use the same/similar settings)?

**Review by jschaul — APPROVED** [review]
Looks good! The proxy hack is funky.

File: docs/reference/elastic-search.md (hunk 3 above)
Comment (akshaymankar) [reply docs/reference/elastic-search.md:143]: I added a `curl` in the comment above the `SHARDS` line.

**Comment by akshaymankar** [comment]
> The proxy hack is funky.

I agree, I am open to suggestions :smile:

**Comment by jschaul** [comment]
> > The proxy hack is funky.
> 
> I agree, I am open to suggestions smile

I think this is good enough in the tests as it is.

**Pull request merged**